### ISSUE TITLE
[REQ] Python file_type streaming upload

### ISSUE THREAD

**Issue by skinkie** [OpenAPITools/openapi-generator#14300]
### Is your feature request related to a problem? Please describe.

When using a large file, python's requests will serialise this to memory. If the form rendering is bigger than the available memory, this will fail.

## Describe the solution you'd like

Analogue to https://pypi.org/project/requests-toolbelt/ a way where file like objects offered via a "open(..., 'rb')" or "open(..., 'rb', buffered=0)" interface are never encoded to a body prior to sending.

## Describe alternatives you've considered

I was hoping it worked out of the box.

**Comment by spacether** [comment]
We are limited by the requests client that we are using
- I think that v2 urllib3 will support streaming uploads, it is still in alpha release once it goes stable it can be switched to
- optionally if you want this sooner, you can write a PR using the requests library for the rest client as it supports streaming uploads

**Comment by skinkie** [comment]
@spacether I think in both cases there should be a PR right? Or does "v2 urllib3" (where can that be found?) support streaming transparently?

**Comment by spacether** [comment]
Yes both would need a PR v2 urllib3 can be found on their releases page: https://github.com/urllib3/urllib3/releases
